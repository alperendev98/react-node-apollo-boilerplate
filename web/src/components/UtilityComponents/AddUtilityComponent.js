import React from 'react';
import PropTypes from 'prop-types';
import { getBEMClasses } from '../../helpers/cssClassesHelper';
import Modal from 'react-modal';

import AxisInput from '../common/AxisInput';
import AxisCheckBox from '../common/AxisCheckBox';
import AxisSelect from '../common/AxisSelect';
import AxisDatePicker from '../common/AxisDatePicker';

import '../../styles/components/addUtilityComponent.css';
import '../../styles/components/common/modal.css';

const BaseClasses = getBEMClasses(['base']);
const ModalClasses = getBEMClasses(['modal']);
const AddUtilityClasses = getBEMClasses(['add-utility']);

class AddUtilityComponent extends React.Component {
	static defaultProps = {
		isAddUtilityModalOpen: false
	};

	static propTypes = {
		isAddUtilityModalOpen: PropTypes.bool
	};

	constructor(props) {
		super(props);
	}

	componentWillMount() {
		Modal.setAppElement('body');
	}

	render() {
		const { isAddUtilityModalOpen, hideAddUtilityModal } = this.props;
		return (
			<React.Fragment>
				<Modal isOpen={isAddUtilityModalOpen} className={AddUtilityClasses('container')}>
					<div className={ModalClasses('title')}>Add Utility</div>
					<div className={ModalClasses('content')}>
						<div className={AddUtilityClasses('row')}>
							<AxisInput label="Utility Name" placeholder="Utility Name" />
						</div>
						<div className={AddUtilityClasses('row')}>
							<AxisSelect label="Metering Type" placeholder="Select Type" />
							<AxisInput label="Average General Service Rate" placeholder="Type Average" />
						</div>
						<div className={AddUtilityClasses('half-row')}>
							<AxisDatePicker label="Metering Type" placeholder="MM/DD/YYYY" />
						</div>
						<div className={AddUtilityClasses('row')}>
							<AxisInput
								label="Other Details Regarding Metering Policy"
								placeholder="Other Details"
							/>
							<AxisInput label="Contact Info" placeholder="Contact Info" />
						</div>
						<div className={AddUtilityClasses('row')}>
							<AxisSelect label="Utility Type" placeholder="Select Utility Type" />
							<AxisInput label="Areas Served" placeholder="Areas" />
						</div>
						<div className={AddUtilityClasses('row')}>
							<AxisInput label="G&T Provider" placeholder="G&T Provider" />
							<AxisInput label="Interconnection Policy" placeholder="Interconnection Policy" />
						</div>
						<div className={AddUtilityClasses('row')}>
							<AxisInput label="Related Documents" placeholder="Document Link" icon="link" />
						</div>
						<div className={AddUtilityClasses('row')}>
							<AxisCheckBox label="Display in utilities list on lead capture form" />
						</div>
					</div>
					<div className={ModalClasses('footer')}>
						<div
							onClick={hideAddUtilityModal}
							className={ModalClasses('btn') + ' ' + ModalClasses('btn-cancel')}
						>
							Cancel
						</div>
						<div className={ModalClasses('btn') + ' ' + ModalClasses('btn-save')}>Save Utility</div>
					</div>
				</Modal>
			</React.Fragment>
		);
	}
}

export default AddUtilityComponent;
