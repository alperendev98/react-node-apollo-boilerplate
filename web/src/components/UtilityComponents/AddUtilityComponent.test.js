import React from 'react';
import { shallow } from 'enzyme';

import AddUtilityComponent from './AddUtilityComponent';

describe('AddUtilityComponent TESTS', () => {
	let wrapper;

	beforeEach(() => {
		wrapper = shallow(<AddUtilityComponent />);
	});

	it('renders AddUtilityComponent without crashing', () => {
		expect(wrapper.length).toEqual(1);
	});

	it('should match previos snapshot for AddUtilityComponent', () => {
		expect(wrapper).toMatchSnapshot();
	});
});
