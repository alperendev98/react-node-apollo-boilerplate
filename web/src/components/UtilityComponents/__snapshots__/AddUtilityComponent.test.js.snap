// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AddUtilityComponent TESTS should match previos snapshot for AddUtilityComponent 1`] = `
<Fragment>
  <Modal
    ariaHideApp={true}
    bodyOpenClassName="ReactModal__Body--open"
    className="add-utility__container"
    closeTimeoutMS={0}
    isOpen={false}
    parentSelector={[Function]}
    portalClassName="ReactModalPortal"
    role="dialog"
    shouldCloseOnEsc={true}
    shouldCloseOnOverlayClick={true}
    shouldFocusAfterRender={true}
    shouldReturnFocusAfterClose={true}
  >
    <div
      className="modal__title"
    >
      Add Utility
    </div>
    <div
      className="modal__content"
    >
      <div
        className="add-utility__row"
      >
        <AxisInput
          icon=""
          label="Utility Name"
          placeholder="Utility Name"
        />
      </div>
      <div
        className="add-utility__row"
      >
        <AxisSelect
          label="Metering Type"
          placeholder="Select Type"
        />
        <AxisInput
          icon=""
          label="Average General Service Rate"
          placeholder="Type Average"
        />
      </div>
      <div
        className="add-utility__half-row"
      >
        <AxisDatePicker
          label="Metering Type"
          placeholder="MM/DD/YYYY"
        />
      </div>
      <div
        className="add-utility__row"
      >
        <AxisInput
          icon=""
          label="Other Details Regarding Metering Policy"
          placeholder="Other Details"
        />
        <AxisInput
          icon=""
          label="Contact Info"
          placeholder="Contact Info"
        />
      </div>
      <div
        className="add-utility__row"
      >
        <AxisSelect
          label="Utility Type"
          placeholder="Select Utility Type"
        />
        <AxisInput
          icon=""
          label="Areas Served"
          placeholder="Areas"
        />
      </div>
      <div
        className="add-utility__row"
      >
        <AxisInput
          icon=""
          label="G&T Provider"
          placeholder="G&T Provider"
        />
        <AxisInput
          icon=""
          label="Interconnection Policy"
          placeholder="Interconnection Policy"
        />
      </div>
      <div
        className="add-utility__row"
      >
        <AxisInput
          icon="link"
          label="Related Documents"
          placeholder="Document Link"
        />
      </div>
      <div
        className="add-utility__row"
      >
        <AxisCheckBox
          label="Display in utilities list on lead capture form"
          objectId=""
        />
      </div>
    </div>
    <div
      className="modal__footer"
    >
      <div
        className="modal__btn modal__btn-cancel"
      >
        Cancel
      </div>
      <div
        className="modal__btn modal__btn-save"
      >
        Save Utility
      </div>
    </div>
  </Modal>
</Fragment>
`;
